src/braket/experimental/autoqasm/pulse/pulse.py

```diff
@@ -115,13 +117,21 @@ def play(frame: Frame, waveform: Waveform) -> None:
     _pulse_instruction("play", frame, waveform)
 
 
-def capture_v0(frame: Frame) -> None:
+def capture_v0(frame: Frame) -> BitVar:
     """Adds an instruction to capture the bit output from measuring the specified frame.
 
     Args:
         frame (Frame): Frame on which the capture operation needs to be performed.
+    Returns:
+        BitVar: A BitVar with the result of the capture.
     """
-    _pulse_instruction("_capture_v0_with_return", frame)
+    # FIXME: This is a hack to get the right classical variable name from the generated sequence.
+    # Fix with the Program refactor.
+    sequence = _pulse_instruction("_capture_v0_with_return", frame)
+    name = sequence._program.stack[0].body[-1].body[0].lvalue.name
+    bit = BitVar(name=name)  # Name gets overwritten
+    bit.name = name
```

Comment:
Unrelated to this PR - but is it expected that the name gets overwritten? Should we update the AutoQASM type `__init__` functions so that they don't overwrite `name` if one is provided in `kwargs`?

Reply:
Yes I already have an issue open for that from a while ago. Trying to fix it lead to a surprising number of errors and it also comes up surprisingly often. Probably a good next item to work on when I have just half a week for AutoQASM